### PR TITLE
restore-util: wait for scatter region sequentially 

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

### What problem does this PR solve? 
Waiting for scatter region sequentially to avoid to create goroutine by RegionSplitter

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: pkg/restore-util/split.go [hunk 1]
```diff
@@ -46,43 +45,54 @@ func NewRegionSplitter(client Client) *RegionSplitter {
...
 		}
-		err = rs.maybeSplitRegion(ctx, rg, &wg)
+		var newRegion *RegionInfo
+		newRegion, err = rs.maybeSplitRegion(ctx, rg)
+		if err == nil {
+			if newRegion != nil {
+				scatterRegions = append(scatterRegions, newRegion)
+			}
+		}
 		return err == nil
```
Comment (kennytm) [inline pkg/restore-util/split.go]:
```go
if err != nil {
    return false
}
if newRegion != nil {
    scatterRegions = append(scatterRegions, newRegion)
}
return true
```

File: pkg/restore-util/client.go [hunk 2]
```diff
@@ -109,6 +109,7 @@ func (c *pdClient) SplitRegion(ctx context.Context, regionInfo *RegionInfo, key
 		return nil, err
 	}
 	conn, err := grpc.Dial(store.GetAddress(), grpc.WithInsecure())
+	defer conn.Close()
```
Comment (overvenus) [inline pkg/restore-util/client.go:112]: Nice catch.

Also, as a future improvement, we can cache the gRPC connection in the pdClient.

File: pkg/restore-util/split.go (hunk 1 above)
Comment (5kbpers) [reply pkg/restore-util/split.go]: oh, I forget to modify it after removing debug log.😂️

**Review by kennytm — COMMENTED** [review]
LGTM

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**